### DISCUSSION TITLE
plot axes padding

### DISCUSSION THREAD

**Discussion by erflot** [cetz-package/cetz-plot#57] [Q&A]
Some extra space at the ends of scatterplot axes to avoid markers sitting on the plot border can be done by computing x-min, x-max, y-min, y-max from data, with some padding added.
Using ```set-style(axes: (padding: 12pt))``` does quite the same, but grid and plot border do not follow the move of axes.
Is there a another setting?

**Comment by johannes-wolf** [comment]
There is no setting to achieve that as of now. Would something like `x-max-offset: <...>` to add a value to an axis `min`/`max` value work?

**Comment by erflot** [comment]
Actually ```set-style(axes: (padding: ...))``` already provides the right padding for all axes at once, but it looks a bit weird with the gaps between axes and plot area. A global setting could be a shortcut to separate offsets for each end of each axis.

**Reply by johannes-wolf** [reply]
Can you give an example for the weird looking gaps please?

**Comment by erflot** [comment]
Sorry for forgetting the demo. 
The first plot shows marks sitting on the axes, the second one is the same plot with ```set-style(axes: (padding: 12pt))```.
Thanks for looking into it. 

![image](https://github.com/user-attachments/assets/ffa5e946-1f3e-4fb6-978f-3ecf07b470ba)

![image](https://github.com/user-attachments/assets/3e2eefb2-7ea3-47aa-9bac-5882ea1b5772)

**Comment by johannes-wolf** [comment]
Thank you for your example. This is actually the expected behaviour of the `padding:` key. I will look into changing it to `outset:` and make `padding:` like the padding you expect.

Note that the next version of cetz' plotting lib is developed in this https://github.com/cetz-package/cetz-plot repository.

**Comment by erflot** [comment]
Thanks for your info.  I have questions for box-whisker plots:
- y-grid and y-ticks are not shown, if y-tick-step is not set

setting the y-tick-step explicitly helps, \
but
- options mark and mark-size for outliers are not used
- label does not trigger the legend like for x-y-scatterplot
- how to set the x-axis to categorical, i.e. showing "A" and "B" at positions 1 and 2

The attached box-whisker.zip has sample.typ and sample.pdf inside.

[box-whisker.zip](https://github.com/user-attachments/files/17362046/box-whisker.zip)
